File: Python/Misc/FastIO.py
import sys, io, os
from io import BytesIO, IOBase
from types import GeneratorType

""" This python file lists ways of getting the input and 
printing the output in much faster way than simple operation
such as following a, b = map(int, input().split()).
"""
_str = str
str = lambda x=b"": x if type(x) is bytes else _str(x).encode()
BUFSIZE = 8192

# There could be more files added in future
class FastIO(IOBase):
    newlines = 0

    def __init__(self, file):
        self._fd = file.fileno()
        self.buffer = BytesIO()
        self.writable = "x" in file.mode or "r" not in file.mode
        self.write = self.buffer.write if self.writable else None

    def read(self):
        while True:
            b = os.read(self._fd, max(os.fstat(self._fd).st_size, BUFSIZE))
            if not b:
                break
            ptr = self.buffer.tell()
            self.buffer.seek(0, 2), self.buffer.write(b), self.buffer.seek(ptr)
        self.newlines = 0
        return self.buffer.read()

    def readline(self):
        while self.newlines == 0:
            b = os.read(self._fd, max(os.fstat(self._fd).st_size, BUFSIZE))
            self.newlines = b.count(b"\n") + (not b)
            ptr = self.buffer.tell()
            self.buffer.seek(0, 2), self.buffer.write(b), self.buffer.seek(ptr)
        self.newlines -= 1
        return self.buffer.readline()

    def flush(self):
        if self.writable:
            os.write(self._fd, self.buffer.getvalue())
            self.buffer.truncate(0), self.buffer.seek(0)

class IOWrapper(IOBase):
    def __init__(self, file):
        self.buffer = FastIO(file)
        self.flush = self.buffer.flush
        self.writable = self.buffer.writable
        self.write = lambda s: self.buffer.write(s.encode("ascii"))
        self.read = lambda: self.buffer.read().decode("ascii")
        self.readline = lambda: self.buffer.readline().decode("ascii")

class FastIO:
    # First Way (Using sys):
    input = sys.stdin.readline
    print = sys.stdout.write
    
    # Second Way (Using io library):
    input = io.BytesIO(os.read(0,os.fstat(0).st_size)).readline
    print = sys.stdout.write(" ".join(map(str,list)) + "\n")
    
    # Third Way (Using the I/O class defined by the user - see the class above)
    sys.stdin, sys.stdout = IOWrapper(sys.stdin), IOWrapper(sys.stdout)
    input = lambda: sys.stdin.readline().rstrip("\r\n")
    output = lambda x: sys.stdout.write(x)

    # Side Track (Use the code below to speed up the recursion using stack)
    def bootstrap(f, stack=[]):
        def wrappedfunc(*args, **kwargs):
            if stack:
                return f(*args, **kwargs)
            else:
                to = f(*args, **kwargs)
                while True:
                    if type(to) is GeneratorType:
                        stack.append(to)
                        to = next(to)
                    else:
                        stack.pop()
                        if not stack:
                            break
                        to = stack[-1].send(to)
                return to
        return wrappedfunc

File: Java/Misc/FastIO.java
package Java.Misc;

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.StringTokenizer;
// Specifically for the Scanner Class
import java.util.Scanner;

// There could be more files added in future
public class FastIO {
  // First Way (Using Scanner Class):
  public static void firstMethod(String[] args) {
    try (Scanner s = new Scanner(System.in)) {
      int n = s.nextInt();
      int k = s.nextInt();
      int count = 0;
      while (n-- > 0) {
          int x = s.nextInt();
          if (x % k == 0)
              count++;
      }
      System.out.println(count);
    }
  }

  // Second Way (Using BufferedReader Class):
  public static void secondMethod(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(
        new InputStreamReader(System.in));

    StringTokenizer st
        = new StringTokenizer(br.readLine());
    int n = Integer.parseInt(st.nextToken());
    int k = Integer.parseInt(st.nextToken());
    int count = 0;
    while (n-- > 0) {
        int x = Integer.parseInt(br.readLine());
        if (x % k == 0)
            count++;
    }
    System.out.println(count);
  }

  // Third Way (User-defined Class) (The fastest)
  static class FastReader {
    BufferedReader br;
    StringTokenizer st;

    public FastReader() {
      br = new BufferedReader(
            new InputStreamReader(System.in));
    }

    String next() {
      while (st == null || !st.hasMoreElements()) {
        try {
          st = new StringTokenizer(br.readLine());
        }
        catch (IOException e) {
          e.printStackTrace();
        }
      }
      return st.nextToken();
    }

    int nextInt() { return Integer.parseInt(next()); }

    long nextLong() { return Long.parseLong(next()); }

    double nextDouble() {
      return Double.parseDouble(next());
    }

    String nextLine() {
      String str = "";
      try {
          if(st.hasMoreTokens()){
            str = st.nextToken("\n");
          }
          else {
            str = br.readLine();
            }
        }
      catch (IOException e) {
        e.printStackTrace();
      }
    return str;
    }
  }

  public static void thirdMethod(String[] args) {
    FastReader s = new FastReader();
    int n = s.nextInt();
    int k = s.nextInt();
    int count = 0;
    while (n-- > 0) {
      int x = s.nextInt();
      if (x % k == 0)
        count++;
    }
    System.out.println(count);
    }
}